src/builtins.c

```diff
@@ -1005,7 +1005,9 @@ JL_CALLABLE(jl_f_invoke)
     jl_value_t *argtypes = args[1];
     JL_GC_PUSH1(&argtypes);
     if (jl_is_tuple(args[1])) {
-        // TODO: maybe deprecation warning, better checking
+        jl_depwarn("`invoke(f, (types...), ...)` is deprecated, "
```

Comment:
stevengj had some comments about this in one of Jameson's recent PR, it's kind of a markdown-ism but one a lot of us have gotten used to